modules/migration-mtc-cr-manifests.adoc

```diff
@@ -0,0 +1,363 @@
+  name: host <1>
+  namespace: openshift-migration
+spec:
+  isHostCluster: <true> <2>
+  azureResourceGroup: <azure_resource_group> <3>
+  caBundle: <ca_bundle_base64> <4>
+  insecure: false <5>
+  refresh: false <6>
+# The 'restartRestic' parameter is relevant for a source cluster.
+# restartRestic: true <7>
+# The following parameters are relevant for a remote cluster.
+# isHostCluster: false
+# exposedRegistryPath: <8>
+# url: <destination_cluster_url> <9>
+# serviceAccountSecretRef:
+#   name: <source_secret_ref> <10>
+#   namespace: openshift-config
+----
+<1> `host` is the default name of the cluster on which the `migration-controller` is running. You can specify a different name.
+<2> `isHostCluster`: The `migration-controller` runs on this cluster if `true`.
+<3> `azureResourceGroup`: If the storage provider is Microsoft Azure, specify the resource group.
+<4> `caBundle`: Base64-encoded certificate bundle for self-signed CA certificates. Required if `insecure` is `false`.
+<5> `insecure`: SSL verification is enabled if `false`. CA certificates are not required or checked if `true`.
+<6> `refresh`: Validates the cluster if `true`.
+<7> `restartRestic`: Restarts `restic` pods on a source cluster after the `stage` pods are created if `true`.
+<8> `exposedRegistryPath`: Exposed registry path of a remote cluster. Required for direct image migration.
+<9> Specify the URL of the remote cluster.
+<10> Specify the name of the `Secret` CR for the remote cluster.
+
+[id="mighook_{context}"]
+== `MigHook`
+
+`MigHook` defines an Ansible playbook or a custom image that runs tasks at a specified stage of the migration.
+
+[source,yaml]
+----
+apiVersion: migration.openshift.io/v1alpha1
+kind: MigHook
+metadata:
+  generateName: <hook_name_prefix> <1>
+  name: <hook_name> <2>
+  namespace: openshift-migration
+spec:
+  activeDeadlineSeconds: <3>
+  custom: false <4>
+  image: <hook_image> <5>
+  playbook: <ansible_playbook_base64> <6>
+  targetCluster: <cluster> <7>
+----
+<1> `generateName`: Optional. If you specify a value for this parameter, for example, `my_hook_`, a unique hash is appended. You do not need to specify the value of the `name` parameter.
+<2> `name`: Migration hook name. Not required if you specify the value of the `generateName` parameter.
+<3> `activeDeadlineSeconds`: Optional. Defines the maximum number of seconds that a hook can run. Default value is `1800`.
+<4> `custom`: The hook is a custom image if `true`. The custom image can include Ansible or it can be written in a different programming language.
+<5> `image`: Specify the custom image, for example, `quay.io/konveyor/hook-runner:latest`. Required if `custom` is `true`.
+<6> `playbook`: Base64-encoded Ansible playbook. Required if `custom` is `false`.
+<7> `targetCluster`: Specify `source` or `destination` as the cluster on which the hook will run.
+
+[id="migmigration_{context}"]
+== `MigMigration`
+
+`MigMigration` runs an associated `MigPlan`.
```

Comment:
So I have no idea what _Mig_ is short for, but can't this be

> _Whatever_ migration runs an associated _whatever_ plan?

Otherwise this needs to follow object conventions, and `MigMigration` object runs an associated `MigPlan` object is an awkward construction.